### PR TITLE
Merging develop into master

### PR DESCRIPTION
**Pull request by mexanick**
Code refactoring, fixing all the issue w.r.t. indentation and dilepton collections
### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

Fixes all the indentation problems, including the one leading to the ProductNotFound error

## How Has This Been Tested?

Tested with CMSSW_8_0_29 by running locally with cmsRun

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: test/DataMCSpectraComparison/histosSimplified.py [hunk 1]
```diff
@@ -13,9 +13,9 @@
 from SUSYBSMAnalysis.Zprime2muAnalysis.Zprime2muAnalysis_cff import goodDataFiltersMiniAOD
 
 process.source.fileNames =[#'file:./pat.root'
-'/TTTo2L2Nu_TuneCUETP8M2_ttHtranche3_13TeV-powheg-pythia8/RunIISummer16MiniAODv2-PUMoriond17_80X_mcRun2_asymptotic_2016_TrancheIV_v6-v1/MINIAODSIM'
+'/store/mc/RunIISummer16MiniAODv2/ZToMuMu_NNPDF30_13TeV-powheg_M_120_200/MINIAODSIM/PUMoriond17_80X_mcRun2_asymptotic_2016_TrancheIV_v6-v1/80000/824C363B-0AC8-E611-B4A5-20CF3027A580.root',
 			   ]
-process.maxEvents.input = -1
+process.maxEvents.input = 100 # Set to a reasonable number (e.g.100) when testing locally with cmsRun
```
Comment (rathjd) [inline test/DataMCSpectraComparison/histosSimplified.py]: Back to 100 in the main code? I'd advise setting this back to -1, or at least remember doing that when it propagates from the development to the main branch.

File: test/DataMCSpectraComparison/histosSimplified.py [hunk 2]
```diff
@@ -278,21 +283,34 @@ def ntuplify(process, fill_gen_info=False):
                                            jet_src = cms.InputTag("slimmedJets"),
                                            beamspot_src = cms.InputTag('offlineBeamSpot'),
                                            vertices_src = cms.InputTag('offlineSlimmedPrimaryVertices'),
-                                           #TriggerResults_src = cms.InputTag('TriggerResults', '', 'PAT'),	#mc
-                                           TriggerResults_src = cms.InputTag('TriggerResults', '', 'RECO'),	#data
+                                           TriggerResults_src = cms.InputTag('TriggerResults', '', 'PAT'),	#mc
```
Comment (rathjd) [inline test/DataMCSpectraComparison/histosSimplified.py]: Can we do this more generically with an isData switch?

File: test/DataMCSpectraComparison/histosSimplified.py (hunk 1 above)
Comment (mexanick) [reply test/DataMCSpectraComparison/histosSimplified.py]: oups... missed it again... will fix

File: test/DataMCSpectraComparison/histosSimplified.py (hunk 2 above)
Comment (mexanick) [reply test/DataMCSpectraComparison/histosSimplified.py]: fair point

**Review by rathjd — COMMENTED** [review]
Thanks for making the setup instructions clearer and fixing the commentation markers from C++ to python logic.
There are still some general improvements that ought to be made, including an automatic switch between MC and data reconstruction.

File: README.md [hunk 3]
```diff
@@ -1,48 +1,45 @@
...
 
-//This is a method for background sample ntuple production on di-electron channel and e-mu channel.
-
-cmsrel CMSSW_8_0_25
+## Setup CMSSW_8_0_29
+It is required for latest JEC
 
+```bash
+cmsrel CMSSW_8_0_29
 cd CMSSW_8_0_25/src
```
Comment (hyeahyun) [inline README.md]: There is a small typo. 
cd CMSSW_8_0_25/src  ->  cd CMSSW_8_0_29/src

File: README.md [hunk 4]
```diff
@@ -1,48 +1,45 @@
...
-
 cmsenv
-
+mkdir SUSYBSMAnalysis && cd SUSYBSMAnalysis/
 git clone git@github.com:BFF-Zprime-mumu/Zprime2muAnalysis.git
```
Comment (hyeahyun) [inline README.md]: Must complete the electron HEEP ID setup before downloading our analysis codes.
If it does not, you get an error corresponding to the # 1 error on the slide that I announced this week.

**Review by hyeahyun — CHANGES_REQUESTED** [review]
I think there is nothing wrong with the codes overall.

**Review by hyeahyun — CHANGES_REQUESTED** [review]
I think there is nothing wrong with the code overall.
I only request changes for minor things in the setup process.

**Comment by mexanick** [comment]
All comments has been addressed, please check @rathjd @hyeahyun

**Review by hyeahyun — APPROVED** [review]
Thank you for your quick feedback.
I will push the codes I fixed as soon as possible.

**Review by rathjd — APPROVED** [review]
Everything is in order now, it seems.

**Pull request merged**